tests/datastore/test_datastore_profile.py

```diff
@@ -98,3 +103,54 @@ def test_from_public_json() -> None:
     )
     profile = DatastoreProfile2Json.create_from_json(public_profile_schema.object)
     assert isinstance(profile, DatastoreProfileV3io), "Not the right profile"
+
+
+class TestTDEngineProfile:
+    @staticmethod
+    def test_from_dsn() -> None:
+        dsn = "taosws://root:taosdata@localhost:6041"
+        profile_name = "test-taosws"
+        profile = TDEngineDatastoreProfile.from_dsn(dsn=dsn, profile_name=profile_name)
+        assert profile.type == "taosws"
+        assert profile.user == "root"
+        assert profile.password == "taosdata"
+        assert profile.host == "localhost"
+        assert profile.port == 6041
+        assert (
+            profile.dsn() == dsn
+        ), "Converting the profile back to DSN did not work as expected"
+
+    @staticmethod
+    def test_datastore_profile_read_from_env(monkeypatch: pytest.MonkeyPatch) -> None:
+        profile_name = "test-profile"
+        project_name = "test-project"
+
+        public_profile = mlrun.common.schemas.DatastoreProfile(
+            name=profile_name,
+            type="taosws",
+            object='{"type":"dGFvc3dz","name":"dGRlbmdpbmUx","user":"cm9vdA==","host":"MC4wLjAuMA==","port":"NjA0MQ=="}',
+            private=None,
+            project=project_name,
+        )
+
+        with patch(
+            "mlrun.db.nopdb.NopDB.get_datastore_profile", return_value=public_profile
+        ):
+            monkeypatch.setenv(
+                f"datastore-profiles.{project_name}.{profile_name}",
+                '{"password": "MTIzNA=="}',
+            )
+            profile_read = datastore_profile_read(f"ds://{profile_name}", project_name)
+
+        assert profile_read.type == "taosws", "Wrong profile type"
+        assert profile_read.password == "1234", "Wrong password"
+
+
+def test_datastore_type_map() -> None:
+    assert set(_DATASTORE_TYPE_TO_PROFILE_CLASS.values()) == set(
```

Comment:
Are you sure this test isn’t too strict? It could be broken simply by introducing an intermediate `DatastoreProfile` class. For instance, you might define `NumericDatastoreProfile(DatastoreProfile)`, then derive `DigitalSafeDataStoreProfile` from `NumericDatastoreProfile` and `AnalogSafeDatastoreProfile` directly from `DatastoreProfile`.

Reply:
Yes, I agree that in this situation, we'll also have to take the deeper level of inheritance.
As we discussed orally, we will leave this for the future :)